libraries/AP_LTM_Telem/AP_LTM_Telem.h

```diff
@@ -0,0 +1,86 @@
+/*
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_SerialManager/AP_SerialManager.h>
+
+#define LTM_PAYLOAD_STATUS_CAPACITY               5    // size of the message buffer queue (max number of messages waiting to be sent)
+
+#define LIGHTTELEMETRY_START1                     0x24 //$
+#define LIGHTTELEMETRY_START2                     0x54 //T
+#define LIGHTTELEMETRY_GFRAME                     0x47 //G GPS + Baro altitude data ( Lat, Lon, Speed, Alt, Sats, Sat fix)
+#define LIGHTTELEMETRY_AFRAME                     0x41 //A Attitude data ( Roll,Pitch, Heading )
+#define LIGHTTELEMETRY_SFRAME                     0x53 //S Sensors/Status data ( VBat, Consumed current, Rssi, Airspeed, Arm status, Failsafe status, Flight mode )
+#define LTM_GFRAME_SIZE                           18
+#define LTM_AFRAME_SIZE                           10
+#define LTM_SFRAME_SIZE                           11
+#define ROLL_LIMIT                                0x7FF
+#define PITCH_LIMIT                               0x3FF
+#define YAW_LIMIT                                 0x7FF
+
+class AP_LTM_Telem {
+public:
+    // Constructor
+    AP_LTM_Telem() {}
+
+    /* Do not allow copies */
+    AP_LTM_Telem(const AP_LTM_Telem &other) = delete;
+    AP_LTM_Telem &operator=(const AP_LTM_Telem&) = delete;
+
+    // init - perform required initialisation
+    void init();
+
+    void update_control_mode(uint8_t mode) {
+        _uav.flightmode = mode;
+    }
+
+private:
+    AP_HAL::UARTDriver *_port;              // UART
+
+    uint8_t _ltm_scheduler;
+    uint32_t _last_frame_ms;
+
+    struct
```

Comment:
This is the state I don't think you need to keep around - exception being the flightmode, of course.

Reply:
Fixed